### DISCUSSION TITLE
How to connect to RabbitMQ cluster?

### DISCUSSION THREAD

**Discussion by MerlinFeng** [rabbitmq/amqp091-go#137] [Q&A] [answered]
Hello,

I have three nodes of RabbitMQ in cluster and want use them properly according to [docs](https://www.rabbitmq.com/clustering.html#clustering-and-clients) but I cannot find any mentions of how to specify their FQDNs properly, it there a way to use clustered RabbitMQ at all?

Thanks in advance.

Old issue is https://github.com/streadway/amqp/issues/339。

**Comment by Zerpet** [comment, marked as answer]
Hi there 👋 We prefer to use GitHub discussions for questions, and use issues only for feature requests and bug reports.

Regarding your question, this library does not provide a mechanism to automatically pick a server from a list. In order to support clustered RabbitMQ, your application should keep a list of FQDNs or URIs to connect to RabbitMQ; then, your application should try to `Dial()` or `DialTLS()` to just one of those URIs. If the client disconnects for any reason, it should pick a different URI to reconnect.
